### DISCUSSION TITLE
JVM memory increase

### DISCUSSION THREAD

**Discussion by yashtandon113** [ant-media/Ant-Media-Server#5043] [FAQs] [answered]
How to increase the JVM memory for Ant Media Server?

**Comment by Mohit-3196** [comment, marked as answer]
Hello,
By default, the JVM  memory is 1/4th of the total memory.

To increase JVM memory, you need to edit and add below property under `/etc/systemd/system/antmedia.service` file.

`Environment=JVM_MEMORY_OPTIONS=-Xmx4g` (This is for 4GB JVM memory. Please change the value as per your requirement)

After saving the file, run below commands

```
sudo systemctl daemon-reload
sudo systemctl antmedia restart
```

**Reply by ghost** [reply]
Hi,
I have installed the antmedia on docker and there was no antmedia.service file in the mentioned location.
How can i increase the heap memory now.?

**Comment by muratugureminoglu** [comment]
Hi @mann9211 

I apologize for the late response. I needed to make some changes to set the -Xms and -Xmx sizes in the Dockerfile and start.sh file.

These changes will be included in the next release, but you can run it with some modifications.

**1.** Download the Ant Media Server zip file and place the [new start.sh](https://github.com/ant-media/Ant-Media-Server/blob/c34255c3550b2c2e5f7471f1e10823f0744d29ce/src/main/server/start.sh) file inside it, then compress it again into a zip file.

**2.** Adjust the `-Xms` and `-Xmx` sizes in the provided [Dockerfile](https://github.com/ant-media/Scripts/blob/1c161291a24873be973c9b6a67f4a7127676f60d/docker/Dockerfile_Process) (Don't forget to uncomment ENV line), then rebuild the Docker image using `docker build`. It will start running with the values you specified.

Reference: https://antmedia.io/docs/guides/clustering-and-scaling/docker/docker-and-docker-compose-installation/

If you need further assistance, don't hesitate to reach out.
